libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -149,38 +150,123 @@
         sb.append(AND).append(CLIENT_ID).append(EQUAL).append(Uri.encode(clientId));
         if (scopes != null && scopes.length > 0)
             sb.append(AND).append(SCOPE).append(EQUAL).append(Uri.encode(computeScopeParameter(scopes)));
         sb.append(AND).append(REDIRECT_URI).append(EQUAL).append(callbackUrl);
+
+        if(addlParams !=null && addlParams.size() > 0) {
+            for(Map.Entry<String,String> entry : addlParams.entrySet()) {
+                String value = entry.getValue()==null?EMPTY_STRING :entry.getValue();
+                sb.append(AND).append(entry.getKey()).append(EQUAL).append(Uri.encode(value));
+            }
+        }
         return URI.create(sb.toString());
     }
 
+    /**
+     * Build the URL to the authorization web page for this login server.
+     * @param loginServer
+     * @param clientId
+     * @param callbackUrl
+     * @param scopes
+     * @param clientSecret
+     * @param displayType
+     * @param accessToken
+     * @param instanceURL
+     * @return
+     */
     public static URI getAuthorizationUrl(URI loginServer, String clientId,
                                           String callbackUrl, String[] scopes, String clientSecret,
                                           String displayType, String accessToken, String instanceURL) {
+        return getAuthorizationUrl(loginServer, clientId, callbackUrl, scopes, clientSecret,
+                displayType, accessToken, instanceURL,null);
+    }
+
+    /**
+     * Build the URL to the authorization web page for this login server.
+     * @param loginServer
+     * @param clientId
+     * @param callbackUrl
+     * @param scopes
+     * @param clientSecret
+     * @param displayType
+     * @param accessToken
+     * @param instanceURL
+     * @param addlParams
+     * @return
+     */
+    public static URI getAuthorizationUrl(URI loginServer, String clientId,
+                                          String callbackUrl, String[] scopes, String clientSecret,
+                                          String displayType, String accessToken, String instanceURL,
+                                          Map<String,String> addlParams) {
         if(accessToken == null || instanceURL == null) {
-            return getAuthorizationUrl(loginServer, clientId, callbackUrl, scopes, clientSecret, displayType);
+            return getAuthorizationUrl(loginServer, clientId, callbackUrl, scopes, clientSecret, displayType, addlParams);
         }
         final StringBuilder sb = new StringBuilder(instanceURL);
         sb.append(FRONTDOOR);
         sb.append(SID).append(EQUAL).append(accessToken);
         sb.append(AND).append(RETURL).append(EQUAL).append(Uri.encode(getAuthorizationUrl(loginServer,clientId,callbackUrl,
                 scopes, clientSecret, displayType).toString()));
+
+        if(addlParams != null && addlParams.size() > 0) {
```

Comment:
The if (addParams != null) {...} is duplicated in two methods - maybe we could have it extracted to its own helper method?